realm/realm-library/src/main/cpp/observable_collection_wrapper.hpp

```diff
@@ -53,57 +52,48 @@ class ObservableCollectionWrapper {
     {
         return m_collection;
     };
-
-    void start_listening(JNIEnv* env, jobject j_collection_object, util::Optional<std::string> subscription_name = util::none);
+    void start_listening(JNIEnv* env, jobject j_collection_object);
     void stop_listening();
 
 private:
-    // Shared logic for creating collection callbacks
-    CollectionChangeCallback create_callback(JNIEnv *env, jobject j_collection_object);
-
     jni_util::JavaGlobalWeakRef m_collection_weak_ref;
     NotificationToken m_notification_token;
     T m_collection;
 };
 
 template <typename T>
-CollectionChangeCallback ObservableCollectionWrapper<T>::create_callback(JNIEnv *env, jobject j_collection_object)
+void ObservableCollectionWrapper<T>::start_listening(JNIEnv* env, jobject j_collection_object)
 {
     static jni_util::JavaClass os_results_class(env, "io/realm/internal/ObservableCollection");
-    static jni_util::JavaMethod notify_change_listeners(env, os_results_class,
-                                                        "notifyChangeListeners", "(J)V");
+    static jni_util::JavaMethod notify_change_listeners(env, os_results_class, "notifyChangeListeners", "(J)V");
 
     if (!m_collection_weak_ref) {
         m_collection_weak_ref = jni_util::JavaGlobalWeakRef(env, j_collection_object);
     }
 
-    bool partial_sync_realm = m_collection.get_realm()->is_partial();
-    auto cb = [=](CollectionChangeSet const &changes, std::exception_ptr err) {
+    auto cb = [=](CollectionChangeSet const& changes, std::exception_ptr err) {
         // OS will call all notifiers' callback in one run, so check the Java exception first!!
         if (env->ExceptionCheck())
             return;
 
-        std::string error_message = "";
         if (err) {
             try {
                 std::rethrow_exception(err);
             }
-            catch (const std::exception &e) {
-                error_message = e.what();
+            catch (const std::exception& e) {
+                realm::jni_util::Log::e("Caught exception in collection change callback %1", e.what());
+                return;
```

Comment:
Can exceptions really be swallowed here?

Reply:
Exceptions in callbacks are really hard to debug since the stack trace will not make much sense.